libexec/scoop-which.ps1

```diff
@@ -19,10 +19,14 @@ $path = "$($gcm.path)"
 $usershims = "$(resolve-path $(shimdir $false))"
 $globalshims = fullpath (shimdir $true) # don't resolve: may not exist
 
-if($path.endswith(".ps1") -and ($path -like "$usershims*" -or $path -like "$globalshims*")) {
-    $shimtext = Get-Content $path
-
-    $exepath = ($shimtext | Where-Object { $_.startswith('$path') }).split(' ') | Select-Object -Last 1 | Invoke-Expression
+if($path -like "$usershims*" -or $path -like "$globalshims*") {
+    $exepath = if ($path.endswith(".exe") -or $path.endswith(".shim")) {
+        (Get-Content "$(join-path ([system.io.fileinfo]$path).DirectoryName ([system.io.fileinfo]$path).BaseName).shim" | Select-Object -First 1).replace('path = ', '')
+    } elseif ($path.endswith(".ps1")) {
+        (Get-Content $path | Where-Object { $_.startswith('$path') }).split(' ') | Select-Object -Last 1 | Invoke-Expression
+    } else {
+        (Select-String -Path $path -Pattern '^(?:@rem|#)\s*(.*)$').Matches.Groups[1].Value
```

Comment:
```suggestion
    } else {
        (Select-String -Path $path -Pattern '^(?:@rem|#)\s*(.*)$').Matches.Groups[1].Value
```

Reply:
Yes, but the PATH comment won't be there in older existing ps1 shims, so I have kept this statement for backward compatibility

Reply:
Done